packages/web/src/components/edit-track/EditTrackModal.tsx

```diff
@@ -155,16 +161,28 @@ const EditTrackModal = ({
   }
 
   const onAddStems = async (selectedStems: File[]) => {
-    const processed = (await Promise.all(processFiles(selectedStems, () => {})))
+    const detectCategory = (filename: string): Nullable<StemCategory> => {
```

Comment:
should these fns be a `useCallback`? should `detectCategory` be defined outside the component so it doesn't get re-created every render?

Reply:
fair